### PR TITLE
Watcher queue now a manager/proxy based faux Queue.

### PR DESCRIPTION
**Pull request by ashcrow**
This change moves the WATCHER_QUEUE to be a faux Queue that uses an
underlying manager based proxy. This fixes issues where the old cache
system was not syncing across some processes.

### CODE REVIEW

**Comment by ashcrow** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;tVngsuMl&#x2F;415-2-multi-host-mgr-move-watcher-queue-to-proxy-manager"> (2) [multi-host-mgr] Move watcher queue to proxy manager</a>

File: src/commissaire/queues.py [hunk 1]
```diff
@@ -62,36 +62,34 @@ def get(self, *args, **kwargs):
...
-                except Empty:
-                    pass
+        obj_model = self._get_obj_model(obj)
+        for x in range(0, len(self._queue)):
+            item = self._queue[x]
+            # If the instance is a tuple snag the first element
+            if isinstance(item, tuple):
+                item = item[0]
+            if obj_model.primary_key == item.primary_key:
+                self._queue.pop(x)
```
Comment (mbarnes) [inline src/commissaire/queues.py]: Popping the queue here will alter the index of subsequent items, no?
Could you iterate over the queue in reverse order?  i.e.
```
for x in reversed(range(len(self._queue))):
```

**Comment by mbarnes** [comment]
I admit I haven't looked closely at the custom watcher queue, but I think I follow the changes here.   Spotted one potential got'cha, but otherwise looks okay.

File: src/commissaire/queues.py (hunk 1 above)
Comment (ashcrow) [inline src/commissaire/queues.py]: Good point. We could also just return out of the loop at that point because we've found what we are looking for.

Comment (mbarnes) [inline src/commissaire/queues.py]: Ah, okay, if you're only expecting one match then it doesn't really matter.

**Comment by ashcrow** [comment]
:arrow_up:

**Comment by mbarnes** [comment]
Merged manually.  Tag away!

**Pull request closed without merging**